### PR TITLE
Add Overwrite Support For `github_repository_file`

### PR DESCRIPTION
**Pull request by jcudit**
Fixes https://github.com/terraform-providers/terraform-provider-github/issues/438

/cc @techdragon @shoekstra @cornfeedhobo

### CODE REVIEW

File: github/resource_github_repository_file_test.go [hunk 1]
```diff
@@ -297,7 +298,46 @@ func TestAccGithubRepositoryFile_committer(t *testing.T) {
...
+		},
+	})
+}
+
 func testAccCheckGithubRepositoryFileExists(n, path, branch string, content *github.RepositoryContent, commit *github.RepositoryCommit) resource.TestCheckFunc {
+	hardCodedRepoName := "test-repo"
+	return testAccCheckGithubRepositoryFileExistsWithRepo(n, path, branch, hardCodedRepoName, content, commit)
+}
+
+func testAccCheckGithubRepositoryFileExistsWithRepo(n, path, branch, repo string, content *github.RepositoryContent, commit *github.RepositoryCommit) resource.TestCheckFunc {
```
Comment (jcudit) [inline github/resource_github_repository_file_test.go:340]: I went down the path of removing references to "test-repo" and leveraging a per-test generated repository instead.  Those changes turned out to break tests so I've opted to leave the existing code paths alone and expose `testAccCheckGithubRepositoryFileExistsWithRepo` instead.  This allows for creating a repository from a template and overwriting its files during the test suite.

**Comment by jcudit** [comment]
Thanks for raising discussion on the update case.  I had not given it too much thought.

After thinking about this a bit, I recommend we rebrand `overwrite` to `overwrite_on_create`.  
We can then assume that whenever the update code path is executed, we will _always_ be overwriting a file, which matches the existing implementation.

To further safeguard, we can add the `ForceNew` attribute to `overwrite_on_create` so that if configuration changes, we can start fresh with a new file.  What do you think @anGie44 ?

**Comment by jcudit** [comment]
q

**Comment by jaxxstorm** [comment]
Is there any news on when this might be reviewed/merged?

**Comment by anGie44** [comment]
> I recommend we rebrand `overwrite` to `overwrite_on_create`.
I like this idea, suits the intended behavior 👍 my only hesitation would be adding the `ForceNew` attribute since the field is intended to be used only at `create` time and it might be more of a pain point to force recreation for users

**Comment by jcudit** [comment]
Tests are [looking good](https://github.com/terraform-providers/terraform-provider-github/runs/1203794419).  The GHES ones are failing as the infrastructure is down at the moment.

**Pull request merged**